python/llm/src/ipex_llm/transformers/lookup.py

```diff
@@ -381,11 +381,20 @@ def lookup_generate(self,
             self.post_time.append(pot-mot)
 
         # Stop on eos and remove content after eos
-        output_ids_list = output_ids[0].tolist()
-        if generation_config.eos_token_id in output_ids_list:
-            idx = output_ids_list.index(generation_config.eos_token_id)
-            step -= (len(output_ids_list) - idx - 1)
-            break
+        if generation_config.eos_token_id is not None:
+            output_ids_list = output_ids[0].tolist()
+            if isinstance(generation_config.eos_token_id, list):
+                eos_token_id_set = set(generation_config.eos_token_id)
+            else:
+                eos_token_id_set = set([generation_config.eos_token_id])
```

Comment:
This can be initialized just once

Reply:
Thank you for the comments. Have move the initialization before the loop.